### PR TITLE
feat(optics): new lens for Redux useState, aka `duo`

### PR DESCRIPTION
**Pull request by serras**
Should I add any new optic? With this (and the resolution of #29, which I leave for contributors) we have Immutable.js and React covered.

### CODE REVIEW

File: __tests__/Lens.test.js [hunk 1]
```diff
@@ -156,4 +156,12 @@ describe('Lens', () => {
   test('should sequence lenses correctly', () => {
     expect(toArray(sequence('id', 'name'), user)).toEqual([1, 'Flavio'])
   })
+
+  test('useState-like lens works', () => {
+    var x = 1
```
Comment (kutyel) [inline __tests__/Lens.test.js]:
```suggestion
    let x = 1
```

File: README.md [hunk 2]
```diff
@@ -254,6 +254,10 @@ Given a key, both `view` and `set`/`over` operate on that key in an object. Howe
 
 Given an index, both `view` and `set`/`over` operate on that index in an array. This lens cannot be used to grow or shrink the array, you can only access or modify positions which are already available.
 
+#### `duo : [v, (v) => nothing] -> Lens s v`
+
+Wraps a two-element pair `[current value, setter for the value]` as a lens. This is useful in combination with React's [`useState`](https://reactjs.org/docs/hooks-reference.html#usestate). Note that in this case the last argument of `view` or `set`, the element you are chaning, is irrelevant.
```
Comment (kutyel) [inline README.md]:
```suggestion
Wraps a two-element pair `[current value, setter for the value]` as a lens. This is useful in combination with React's [`useState`](https://reactjs.org/docs/hooks-reference.html#usestate). Note that in this case the last argument of `view` or `set`, the element you are changing, is irrelevant.
```

**Review by kutyel — CHANGES_REQUESTED** [review]
Should we add new optics? Does `duo` exist? 🤔  Never heard of ti 😂

File: __tests__/Lens.test.js (hunk 1 above)
Comment (serras) [reply __tests__/Lens.test.js]: The point here is being able to check that the value has changed, so we really need `var`.

**Comment by serras** [comment]
Oh, you can totally change the name! The point was to have a way to create a lens for use in React. Now you can do:

```js
const thing = duo(useState(0))
// ... code in between ... 
over(thing, x => x + 1, {})  // calls useState with 1
```

File: __tests__/Lens.test.js (hunk 1 above)
Comment (kutyel) [reply __tests__/Lens.test.js]: You should *never* use `var` now in modern JS, if you want mutable variables, use `let` 😉

**Review by kutyel — APPROVED** [review]
🚀

**Pull request merged**